central/image/service/service_impl_test.go

```diff
@@ -3,9 +3,51 @@ package service
 import (
 	"testing"
 
+	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/grpc/testutils"
+	"github.com/stretchr/testify/assert"
 )
 
 func TestAuthz(t *testing.T) {
 	testutils.AssertAuthzWorks(t, &serviceImpl{})
 }
+
+func TestBuildNames(t *testing.T) {
+	srcImage := &storage.ImageName{FullName: "si"}
+
+	t.Run("nil metadata", func(t *testing.T) {
+		names := buildNames(srcImage, nil)
+		assert.Len(t, names, 1)
+		assert.Equal(t, srcImage.GetFullName(), names[0].GetFullName())
+	})
+
+	t.Run("empty metadata", func(t *testing.T) {
+		names := buildNames(srcImage, &storage.ImageMetadata{})
+		assert.Len(t, names, 1)
+		assert.Equal(t, srcImage.GetFullName(), names[0].GetFullName())
+	})
+
+	t.Run("metadata with data source", func(t *testing.T) {
+		metadata := &storage.ImageMetadata{DataSource: &storage.DataSource{}}
+		names := buildNames(srcImage, metadata)
+		assert.Len(t, names, 1)
+		assert.Equal(t, srcImage.GetFullName(), names[0].GetFullName())
+	})
+
+	t.Run("metadata with mirror", func(t *testing.T) {
+		mirror := "example.com/mirror/image:latest"
+		metadata := &storage.ImageMetadata{DataSource: &storage.DataSource{Mirror: mirror}}
+		names := buildNames(srcImage, metadata)
+		assert.Len(t, names, 2)
+		assert.Equal(t, srcImage.GetFullName(), names[0].GetFullName())
+		assert.Equal(t, mirror, names[1].GetFullName())
```

Comment:
That doesn't work.  `names` is of type `[]*storage.ImageName` - if it was of type `[]string` I'd agree.

Eventually I expect `buildNames` to expand to include the names from `existingImg` as well (to better support signatures).  

This could be personal preference, but it may be more readable / maintainable if each are on separate line items vs. editing the middle of long line.